src/core/src/core_logic/MaintenanceWindow.py

```diff
@@ -71,3 +71,27 @@ def is_package_install_time_available(self, remaining_time_in_minutes=None):
         else:
             self.composite_logger.log_warning("Time Remaining: " + str(timedelta(seconds=int(remaining_time_in_minutes * 60))) + ", Cutoff time: " + str(timedelta(minutes=cutoff_time_in_minutes)) + " [Out of time!]")
             return False
+
+    def get_percentage_maintenance_window_used(self):
+        """Calculate percentage of maintenance window used"""
+        try:
+            current_time = self.env_layer.datetime.datetime_utcnow()
+            start_time = self.env_layer.datetime.utc_to_standard_datetime(self.start_time)
+            if current_time < start_time:
+                raise Exception("Start time {} is greater than current time {}".format(str(start_time), str(current_time)))
```

Comment:
This exception will go away, check the other comment on L89, but for future reference and if it's used elsewhere, you need to use a placeholder in the string while using format() 
eg:  "Start time {**0**} is greater than current time {**1**}".format(str(start_time), str(current_time))

Reply:
We do need to throw exception if current_time < start_time because it will not be thrown otherwise. This function will return  negative value but not throw exception if we don't throw exception.